### PR TITLE
Add functionality to handle custom error responses

### LINKED ISSUE #26: Custom response
Currently you will always get a JSON in the following schema if you exceed the rate limit:
```json
{
  "error": "Too Many Requests",
  "message": "Rate limit exceeded, retry in 1 minute",
  "statusCode": 429
}
```

Since I use a different JSON schema in my response, it would be great to have a way of customizing the response. E.g. Having a callback to format/send the response.

### PR DESCRIPTION
**Pull request by pc-jedi**
Introduced new parameter `errorMessage` which is a callback to build the custom error response. It has currently to parameters `after` and `max`. It might be also good to provided `req`, but I want to first have simple solution.
If the callback is not provided it defaults to:
```js
function (after, max) {
  return { statusCode: 429, error: 'Too Many Requests', message: `Rate limit exceeded, retry in ${after}` }
}
```

To make this work I had to remove the serialization. Not sure if it would be better to let the user provided this also als a parameter.

Fixes #26 
Closes #44

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -128,16 +123,11 @@ function buildRouteRate (pluginComponent, params, routeOptions) {
           params.onExceeded(req)
         }
 
-        res.type('application/json').serializer(serializeError)
```
Comment (Eomm) [inline index.js]: I think we could keep this line adding a bool to know if the `errorMessage` has been overwritten by user or not.
This should restore also the changes in tests `t.strictEqual(res.headers['content-type'], 'application/json; charset=utf-8')`

File: README.md [hunk 2]
```diff
@@ -82,6 +85,20 @@ fastify.register(require('fastify-rate-limit'), {
 })
 ```
 
+`errorMessage` example usage:
+```js
+fastify.register(require('fastify-rate-limit'), {
+  /* ... */
+  errorMessage: function(after, max) {
```
Comment (delvedor) [inline README.md]: I would directly add `req` as a first parameter, or change the approach a single object, such as `context`:
```js
{
  after: String,
  max: Number
}
```
The second approach is more future proof :)

Comment (pc-jedi) [reply README.md]: `req` does not contain the `after` or `max`, but might be also needed/nice to have for building error messages.
Is it then better to have to parameter `(req, context)` or shall `req` be part of `context`?
```ts
{
  req: fastify.FastifyRequest<http.IncomingMessage>,
  after: string,
  max: number
}
```

Comment (pc-jedi) [reply README.md]: @delvedor ping :-)

**Comment by allevo** [comment]
I don't like `errorMessage`: the value is a function not a message or string. What do you think?

**Comment by pc-jedi** [comment]
@allevo You're right, me neither. What would be a good alternative?
I want to make clear that this function is only been called if the limit is reached (429).
Then the return value of the function is been used as response body.

**Comment by allevo** [comment]
Maybe `getResponseBody`? But I'm not good in naming...

**Comment by Eomm** [comment]
What about errorMessageFactory?

**Comment by pc-jedi** [comment]
I checked with [fastify docs](https://www.fastify.io/docs/v1.13.x/Server/#seterrorhandler).
There it is `errorHandler`.
But I think `errorResponseFactory` is also a good option.

**Comment by Eomm** [comment]
> I checked with [fastify docs](https://www.fastify.io/docs/v1.13.x/Server/#seterrorhandler).
> There it is `errorHandler`.
> But I think `errorResponseFactory` is also a good option.

Since you are not managing the error I think errorHandler could be confusing (the user may expect to ignore some error for example, but it is mandatory to return a string).

So I would choose between:

- `errorMessage` but it could be a static string or a function (with a complete params signature as suggeste by delvedor)
- `errorWhateverFactory/Builder` in order to explain intuitively that it stands for

File: README.md [hunk 3]
```diff
@@ -56,6 +57,7 @@ fastify.register(require('fastify-rate-limit'), {
   redis: new Redis({ host: '127.0.0.1' }), // default null
   skipOnError: true, // default false
   keyGenerator: function(req) { /* ... */ }, // default (req) => req.raw.ip
+  errorMessage: function(after, max) { /* ... */}, // default (after) => ({ /* ... */ })
```
Comment (Eomm) [inline README.md]:
```suggestion
  errorResponseBuilder: function(req, context) { /* ... */}, 
```

**Pull request merged**